### PR TITLE
 🌱 Adding to the test framework the equivalent to kubectl create -f.

### PR DESCRIPTION
**Pull request by adilGhaffarDev**
**What this PR does / why we need it**:
Adding runtime controller create for testing.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: test/framework/cluster_proxy.go [hunk 1]
```diff
@@ -247,6 +253,27 @@ func (p *clusterProxy) Apply(ctx context.Context, resources []byte, args ...stri
...
+	Expect(ctx).NotTo(BeNil(), "ctx is required for Create")
+	Expect(resources).NotTo(BeNil(), "resources is required for Create")
+
+	var retErrs []error
+	objs, err := yaml.ToUnstructured(resources)
+	if err != nil {
+		return err
+	}
+	for _, o := range objs {
+		if err := p.GetClient().Create(ctx, &o); err != nil {
```
Comment (furkatgofurov7) [inline test/framework/cluster_proxy.go:267]:
```suggestion
	for o := range objs {
		if err := p.GetClient().Create(ctx, &objs[o]); err != nil {
```

or let's reassign the iteration var
```
for _, o := range objs { 
   o := o
   if err := p.GetClient().Create(ctx, &o); err != nil {
```

**Comment by adilGhaffarDev** [comment]
4/4 pull-cluster-api-e2e-full-release-1-4 passed (last passed: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/9731/pull-cluster-api-e2e-full-release-1-4/1725501937413001216)
/test pull-cluster-api-e2e-full-release-1-4

**Comment by furkatgofurov7** [comment]
@adilGhaffarDev can you please remove WIP, if this is ready for review? Thanks!
The last run passed (5/5), re-running it again

/test pull-cluster-api-e2e-full-release-1-4
/area e2e-testing

**Comment by adilGhaffarDev** [comment]
5/6 passed, last failure was not at `Applying the cluster template yaml to the cluster` it failed at `Running Post-upgrade steps against the management cluster`, (https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api/9731/pull-cluster-api-e2e-full-release-1-4/1725625498903515136). This failure is not related to this PR. I will keep on investigating it. 
running once more : 
/test pull-cluster-api-e2e-full-release-1-4

**Comment by adilGhaffarDev** [comment]
5/7 passed last failure was also related to ownerGraph issue which is in  `Running Post-upgrade steps against the management cluster` Looking into it.
/test pull-cluster-api-e2e-full-release-1-4

File: test/e2e/kcp_remediations.go [hunk 2]
```diff
@@ -472,7 +472,7 @@ func createWorkloadClusterAndWait(ctx context.Context, input createWorkloadClust
 	Expect(workloadClusterTemplate).ToNot(BeNil(), "Failed to get the cluster template")
 
 	Eventually(func() error {
-		return input.Proxy.Apply(ctx, workloadClusterTemplate)
+		return input.Proxy.Create(ctx, workloadClusterTemplate)
 	}, 10*time.Second).Should(Succeed(), "Failed to apply the cluster template")
```
Comment (chrischdi) [inline test/e2e/kcp_remediations.go:476]:
```suggestion
	}, 10*time.Second).Should(Succeed(), "Failed to create the cluster template")
```

**Comment by adilGhaffarDev** [comment]
> Q: what is the reason to use create instead of apply?

As @killianmuldoon mentioned here https://github.com/kubernetes-sigs/cluster-api/issues/9696, it will help debug the flake we see in upgrade tests more specifically the one we are getting at `INFO: Applying the cluster template yaml to the cluster`. Here is a further explanation of 3 flakes that we are getting in clusterctl upgrade: https://github.com/kubernetes-sigs/cluster-api/issues/9688#issuecomment-1818939673 

Why did we add this to 1.4 instead of the main first?
Because on 16th Nov, Thursday v1.4 was failing continuously on clusterctl upgrade. I added this to run some tests and get some logs to investigate, but after running several tests on this PR I don't see any failure at `INFO: Applying the cluster template yaml to the cluster`. I will run a few more tests just to confirm.

**Comment by sbueringer** [comment]
This is the same PR as https://github.com/kubernetes-sigs/cluster-api/pull/9736/files just for release-1.4, right?

If yes, let's please hold this PR and continue with the other one. If we merged 9736 we should then cherry-pick back from there

**Comment by chrischdi** [comment]
And please add `[release-1.4]` to the title of the PR :-)

**Pull request closed without merging**